app/controllers/concerns/collaborator_processing.rb

```diff
@@ -23,11 +24,24 @@ def add_users_as_collaborators(resource, user_ids)
     end
   end
 
+  def add_group_members_as_collaborators(resource, group_ids)
+    group_ids.split(',').each do |group_id|
+      add_users_as_collaborators(resource, group_user_ids(group_id), group_id)
+      associate_group_to_resource(group_id, resource)
+    end
+  end
+
   def remove_collaborator(collaborator)
     authorize!(collaborator, 'destroy?')
     collaborator.destroy
   end
 
+  def remove_group_collaborators(collaborators)
+    collaborators.each do |collaborator|
+      remove_collaborator(collaborator)
+    end
```

Comment:
Going back and forth on whether this would make things clearer or more confusing to troubleshoot...